vector/src/main/java/im/vector/app/core/pushers/VectorMessagingHelper.kt

```diff
@@ -0,0 +1,190 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.pushers
+
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import android.os.Looper
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.ProcessLifecycleOwner
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import im.vector.app.core.di.ActiveSessionHolder
+import im.vector.app.core.network.WifiDetector
+import im.vector.app.core.pushers.model.PushData
+import im.vector.app.core.resources.BuildMeta
+import im.vector.app.features.notifications.NotifiableEventResolver
+import im.vector.app.features.notifications.NotificationActionIds
+import im.vector.app.features.notifications.NotificationDrawerManager
+import im.vector.app.features.settings.VectorDataStore
+import im.vector.app.features.settings.VectorPreferences
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.matrix.android.sdk.api.extensions.tryOrNull
+import org.matrix.android.sdk.api.logger.LoggerTag
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.getRoom
+import org.matrix.android.sdk.api.session.room.getTimelineEvent
+import timber.log.Timber
+import javax.inject.Inject
+
+private val loggerTag = LoggerTag("Push", LoggerTag.SYNC)
+
+/**
+ * Hilt injection happen at super.onReceive().
+ */
+
+class VectorMessagingHelper @Inject constructor(
+    private val notificationDrawerManager: NotificationDrawerManager,
+    private val notifiableEventResolver: NotifiableEventResolver,
+    private val activeSessionHolder: ActiveSessionHolder,
+    private val vectorPreferences: VectorPreferences,
+    private val vectorDataStore: VectorDataStore,
+    private val wifiDetector: WifiDetector,
+    private val unifiedPushHelper: UnifiedPushHelper,
+    private val pushParser: PushParser,
+    private val actionIds: NotificationActionIds,
+    private val context: Context,
+    private val buildMeta: BuildMeta
+) {
+
+    private val coroutineScope = CoroutineScope(SupervisorJob())
+
+    // UI handler
+    private val mUIHandler by lazy {
+        Handler(Looper.getMainLooper())
+    }
+
+    /**
+     * Called when message is received.
+     *
+     * @param message the message
+     */
+    fun onMessage(message: String) {
```

Comment:
This could be defined in an interface with this single method.

Reply:
I am not sure how to handle the injections with the interface.